dev/start

```diff
@@ -1,45 +1,164 @@
+_DEVENV="$( cd "${_DEVENV}" 2>&1 > /dev/null ; pwd -P)"
+_PYTHON="${_PYTHON:-3.8}"
+_UPDATE="${_UPDATE:-1}"
+
+# deactivate any prior envs
+if ! [ "${CONDA_SHLVL:-0}" = "0" ]; then
+    echo "Deactivating ${CONDA_SHLVL} environments..."
+    while ! [ "${CONDA_SHLVL:-0}" = "0" ]; do
+        if ! conda deactivate; then
+            echo "Error: failed to deactivate environment" 1>&2
+            return 1
         fi
-        bash miniconda3.sh -bfp "$DEVENV"
-        rm -f miniconda3.sh
-        "${_CONDA}" install -yq -p "$DEVENV" python="$PYTHON" --file tests/requirements.txt -c defaults
-        "${_CONDA}" install -yq -p "$DEVENV" patchelf  # for conda-build
-    else
-        if [ ! -f "miniconda3.exe" ]; then
-            powershell.exe -Command "(new-object System.Net.WebClient).DownloadFile('https://repo.anaconda.com/miniconda/Miniconda3-latest-Windows-x86_64.exe','miniconda3.exe')"
+    done
+fi
+
+# get/set variables
+_STUB="devenv-${_PYTHON}"
+_ENV="${_DEVENV}/envs/${_STUB}"
+_UPDATED="${_ENV}/.devenv-updated"
+case "$(uname)" in
+    Darwin|Linux) _CONDA_EXE="${_DEVENV}/bin/conda" ; _ENV_EXE="${_ENV}/bin/conda" ;;
+               *) _CONDA_EXE="${_DEVENV}/Scripts/conda" ; _ENV_EXE="${_ENV}/Scripts/conda" ;;
+esac
+
+# does conda install exist?
+if ! [ -f "${_DEVENV}/conda-meta/history" ]; then
+    # downloading conda
+    if ! [[ "$(uname)" == "Darwin" && -f "${_DEVENV}/miniconda.sh" ]] && \
+        ! [[ "$(uname)" == "Linux" && -f "${_DEVENV}/miniconda.sh" ]] && \
+        ! [[ -f "${_DEVENV}/miniconda.exe" ]]; then
+        echo "Downloading conda..."
+        case "$(uname)" in
+            Darwin) curl -s "https://repo.anaconda.com/miniconda/Miniconda3-latest-MacOSX-x86_64.sh" -o "${_DEVENV}/miniconda.sh" ;;
+             Linux) curl -s "https://repo.anaconda.com/miniconda/Miniconda3-latest-Linux-x86_64.sh" -o "${_DEVENV}/miniconda.sh" ;;
+                 *) powershell.exe -Command "Invoke-WebRequest -Uri 'https://repo.anaconda.com/miniconda/Miniconda3-latest-Windows-x86_64.exe' -OutFile '${_DEVENV}/miniconda.exe' | Out-Null" ;;
+        esac
+        if ! [ $? = 0 ]; then
+            echo "Error: failed to download conda" 1>&2
+            return 1
         fi
-        cmd.exe /c "start /wait \"\" miniconda3.exe /InstallationType=JustMe /RegisterPython=0 /AddToPath=0 /S /D=%CD%\$(cygpath -w $DEVENV)"
-        _CONDA="$DEVENV/Scripts/conda"
-        "$DEVENV/Scripts/conda" install -yq -p "$DEVENV" python="$PYTHON" --file tests/requirements.txt -c defaults
-        rm -f miniconda3.exe
+    fi
+
+    # installing conda
+    echo "Installing conda..."
+    case "$(uname)" in
+        Darwin|Linux) bash "${_DEVENV}/miniconda.sh" -bfp "${_DEVENV}" > /dev/null ;;
+                   *) cmd.exe /c "start /wait \"\" \"${_DEVENV}\miniconda.exe\" /InstallationType=JustMe /RegisterPython=0 /AddToPath=0 /S /D=${_DEVENV} > NUL" ;;
+    esac
+    if ! [ $? = 0 ]; then
+        echo "Error: failed to install conda" 1>&2
```

Comment:
```suggestion
        echo "Error: failed to create platform base environment" 1>&2
```
Not sure what you prefer, but we don't really install conda but call the miniconda bootstrapper here.